### DISCUSSION TITLE
Running as root - rust (rustup/cargo commands are failing)

### DISCUSSION THREAD

**Discussion by dyasny** [machulav/ec2-github-runner#155] [Q&A]
I've tried using the action today, it provisions an ubuntu aarch64 VM and runs on it. And here's where I ran into serious weirdness. My regular self hosted runners, also ubuntu 22.04, are using a simple manual setup:
- install git, curl, jq, build-essentials and so forth
- run rustup for the ubuntu user

Later on, I use the regular dtolnay/rust-toolchain action, which has always been rock solid, I use it on macs, windows, a bunch of different Linux distros - no problem, until this. On this runner, the action fails because it cannot run the `rustup` command, even though it is present in the system. 

Digging in a bit deeper I found that the action runs on the VM as root, not as the ubuntu user, in which I put the unpacked github actions binaries. 

OK, so if this is root, I also installed rustup for root, and in the base image, I can run rustup and cargo as root and as ubuntu, no problem. The action keeps failing, not being able to find the commands in the $PATH. 

So two things:
1. How do I switch away from running as root? I'd rather stick to the ubuntu user, which has passwordless sudo if needed
2. I'd love to have some insight into how commands are run on the host, the env is weird, with the $PATH not getting affected by changes pushed to $GITHUB_PATH at all.

**Comment by tux-admin** [comment]
I realize this is an old post but I was wondering if you had ever found a good way to work around this issue. All I can come up with at this point is to call the user with sudo -u username -i to get the profile to source.
